### PR TITLE
Add function 'get_pending_files' to FossID API service

### PR DESCRIPTION
**Pull request by nnobelis**
Signed-off-by: Nicolas Nobelis <nicolas.nobelis@bosch.io>

### CODE REVIEW

File: clients/fossid-webapp/src/main/kotlin/Extensions.kt [hunk 1]
```diff
@@ -253,3 +253,16 @@ suspend fun FossIdRestService.listIgnoredFiles(user: String, apiKey: String, sca
             "get_ignored_files", SCAN_GROUP, user, apiKey, "scan_code" to scanCode
         )
     )
+
+/**
+ * List the files of a scan that are in "pending identification" state
+ * @param user the user querying the API
+ * @param apiKey the key to query the API
+ * @param scanCode the code of the scan
```
Comment (sschuberth) [inline clients/fossid-webapp/src/main/kotlin/Extensions.kt:261]: Side note: This is fine for now as it aligns with the existing code in this file. But at some point we should change all function docs to use the preferred KDoc style of parameter references inlined into fluent text.

**Review by sschuberth — APPROVED** [review]

**Review by oheger-bosch — APPROVED** [review]
The changes are fine with me.

There is one point - independent on this PR - which concerns me a bit: As the API of FossID is more of a RPC than a REST API, I ask myself whether Retrofit is actually a good choice for the client implementation. Maybe, with a custom client implementation - after all just a single generic HTTP request has to be implemented -, some aspects and also the usage of the service from client code could be simplified. (But this is a topic for a different discussion.)

File: clients/fossid-webapp/src/main/kotlin/Extensions.kt (hunk 1 above)
Comment (nnobelis) [reply clients/fossid-webapp/src/main/kotlin/Extensions.kt:261]: Ok. When this one in merged I will do a PR for it.

**Comment by nnobelis** [comment]
> The changes are fine with me.
> 
> There is one point - independent on this PR - which concerns me a bit: As the API of FossID is more of a RPC than a REST API, I ask myself whether Retrofit is actually a good choice for the client implementation. Maybe, with a custom client implementation - after all just a single generic HTTP request has to be implemented -, some aspects and also the usage of the service from client code could be simplified. (But this is a topic for a different discussion.)

You are of course right. At the time of the first implementation, we chose to stick with Retrofit to speed up the implementation.
However in the future, we have been told that FossID plans on updating their API so we decided to be conservative and wait for such an update before refactoring the service.

**Pull request merged**